FE-service/src/components/subpages/edition/forms/EditPokemonForm.jsx

```diff
@@ -0,0 +1,193 @@
+import * as z from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { useState, useEffect } from "react";
+import { createPortal } from "react-dom";
+import {
+  TextField,
+  Button,
+  Typography,
+  CircularProgress,
+  FormHelperText,
+} from "@mui/material";
+import {
+  ModalOverlay,
+  ModalContent,
+  FormContent,
+  PokemonName,
+} from "../EditionPage.styles";
+import { useTheme } from "../../../../context/ThemeContext";
+import useDataHandler from "../../../../hooks/useDataHandler";
+import usePokemonApi from "../../../../hooks/usePokemonApi";
+import useGetDbData from "../../../../hooks/useGetDbData";
+import { POKEMON_API_POKEMON, POKEMON_IMG } from "../../../../../../apiConfig";
+import { enqueueSnackbar } from "notistack";
+
+const schema = z.object({
+  height: z.number().positive("Height must be a positive number"),
+  weight: z.number().positive("Weight must be a positive number"),
+  base_experience: z
+    .number()
+    .positive("Base Experience must be a positive number"),
+});
+
+const EditPokemonForm = ({ pokemonId, onClose, onPokemonUpdate }) => {
+  const { theme } = useTheme();
+  const [pokemonName, setPokemonName] = useState("");
+  const { putData, postData } = useDataHandler("pokemons");
+  const {
+    data: pokemonData,
+    loading: pokemonLoading,
+    error: pokemonError,
+  } = useGetDbData("pokemons");
+  const {
+    data: pokemon,
+    loading: apiLoading,
+    error: apiError,
+  } = usePokemonApi(`pokemon/${pokemonId}`);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+  } = useForm({
+    resolver: zodResolver(schema),
+  });
+
+  useEffect(() => {
+    if (pokemonData && pokemonData.length > 0) {
+      const existingPokemon = pokemonData.find(
+        (p) => p.id === String(pokemonId)
+      );
+      if (existingPokemon) {
+        setPokemonName(existingPokemon.name);
+        setValue("height", existingPokemon.height);
+        setValue("weight", existingPokemon.weight);
+        setValue("base_experience", existingPokemon.base_experience);
+      } else {
+        setPokemonName(pokemon.name);
+        setValue("height", pokemon.height);
+        setValue("weight", pokemon.weight);
+        setValue("base_experience", pokemon.base_experience);
+      }
+    }
+  }, [pokemonData, pokemon, setValue, pokemonId]);
+
+  useEffect(() => {
+    if (apiError) {
+      enqueueSnackbar(`Error loading Pokémon data: ${apiError}`, {
```

Comment:
Such an error should be handled within the request. The fewer effects in React, the better.